server/controllers/users.js

```diff
@@ -51,31 +51,36 @@ export const exists = async (req, res) => {
 
 /**
  * Login or create a new user
+ *
+ * TODO: we are passing createProfile from frontend to specify if we need to
+ * create a new account. In the future once signin.js is fully deprecated (replaced by signinV2.js)
+ * this function should be refactored to remove createProfile.
  */
-export const signin = (req, res, next) => {
-  const { user, redirect, websiteUrl } = req.body;
-  let loginLink;
-  let clientIP;
-  return models.User.findOne({ where: { email: user.email.toLowerCase() } })
-    .then(u => u || models.User.createUserWithCollective(user))
-    .then(u => {
-      loginLink = u.generateLoginLink(redirect || '/', websiteUrl);
-      clientIP = req.ip;
-      if (config.env === 'development') {
-        logger.info(`Login Link: ${loginLink}`);
-      }
-      return emailLib.send('user.new.token', u.email, { loginLink, clientIP }, { sendEvenIfNotProduction: true });
-    })
-    .then(() => {
-      const response = { success: true };
-      // For e2e testing, we enable testuser+(admin|member)@opencollective.com to automatically receive the login link
-      if (config.env !== 'production' && user.email.match(/.*test.*@opencollective.com$/)) {
-        response.redirect = loginLink;
-      }
-      return response;
-    })
-    .then(response => res.send(response))
-    .catch(next);
+export const signin = async (req, res) => {
+  const { user, redirect, websiteUrl, createProfile = true } = req.body;
+  let profile = await models.User.findOne({ where: { email: user.email.toLowerCase() } });
+  if (!profile && !createProfile) {
+    return res.send({
```

Comment:
Here, I am not returning an HTTP error code. Just the response with an error object. And on the front-end we check for the `EMAIL_DOES_NOT_EXIST` and if found show the appropriate page. Since this error is specific to open collective internal logic, I didn't see the need to use an HTTP error code; in essence the request was successfully processed; and we return the fact that the email is not found. Do you think we should use an HTTP error code? 🤔

Reply:
GraphQL gives us the habit of not using HTTP error codes, but in the old days of the Internet that would probably have been a 404 - resource not found :-) @znarf wdyt?

Reply:
400 should do the job, I would not do 404 unless the URL is non existing

Reply:
@Betree : So you think we should return an HTTP error code ? My understanding is that Apollo docs discourage this and rather expects to send a custom error code which I have done; Reference: https://www.apollographql.com/docs/apollo-server/data/errors/#returning-http-status-codes

> We recommend using the included [Error Codes](https://www.apollographql.com/docs/apollo-server/data/errors/#error-codes) or [Custom Errors](https://www.apollographql.com/docs/apollo-server/data/errors/#custom-errors) for error consistency rather than directly modifying the HTTP response.

Reply:
@SudharakaP We're not doing Apollo here, it's a REST endpoint - not GraphQL

Reply:
Ops sorry, I had some confusion here; I've added the HTTP 400 code and corrected the front-end as well;

https://github.com/opencollective/opencollective-frontend/pull/7832/commits/f3f0a28ef1ec42dfdcb93f1f10e2255d3d1dd272